src/core/core.scale.js

```diff
@@ -916,28 +932,27 @@ module.exports = Element.extend({
 
 		if (gridLines.drawBorder) {
 			// Draw the line at the edge of the axis
-			context.lineWidth = helpers.valueAtIndexOrDefault(gridLines.lineWidth, 0);
-			context.strokeStyle = helpers.valueAtIndexOrDefault(gridLines.color, 0);
-			var x1 = me.left;
-			var x2 = me.right + axisWidth;
-			var y1 = me.top;
-			var y2 = me.bottom + axisWidth;
-
-			var aliasPixel = helpers.aliasPixel(context.lineWidth);
+			var firstLineWidth = axisWidth;
+			var lastLineWidth = helpers.valueAtIndexOrDefault(gridLines.lineWidth, ticks.length - 1, 0);
+			var x1 = helpers.alignLinePixel(chart, me.left, firstLineWidth) - firstLineWidth / 2;
+			var x2 = helpers.alignLinePixel(chart, me.right, lastLineWidth) + lastLineWidth / 2;
+			var y1 = helpers.alignLinePixel(chart, me.top, firstLineWidth) - firstLineWidth / 2;
+			var y2 = helpers.alignLinePixel(chart, me.bottom, lastLineWidth) + lastLineWidth / 2;
+
 			if (isHorizontal) {
-				y1 = y2 = options.position === 'top' ? me.bottom : me.top;
-				y1 += aliasPixel;
-				y2 += aliasPixel;
+				y1 = y2 = position === 'top' ? alignedBottom : alignedTop;
```

Comment:
I think we are computing too many values here and based on my previous comment, I think we can rewrite this part as:

```javascript
var x1, x2, y1, y2;
if (isHorizontal) {
    x1 = alignPixel(chart, me.left, firstLineWidth) - firstLineWidth / 2;
    x2 = alignPixel(chart, me.right, lastLineWidth) + lastLineWidth / 2;
    y1 = y2 = aligned;
} else {
    y1 = alignPixel(chart, me.top, firstLineWidth) - firstLineWidth / 2;
    y2 = alignPixel(chart, me.bottom, lastLineWidth) + lastLineWidth / 2;
    x1 = x2 = aligned;
}
```